### PR TITLE
Di dev

### PR DESCRIPTION
**Pull request by HoVanDi**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "todolist",
```
Comment (hoihmyit) [inline package.json]: @HoVanDi - pls rename this to "crud_app", not "todolist"

File: src/app/Admin/page.tsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const Todolist = () => {
```
Comment (hoihmyit) [inline src/app/Admin/page.tsx]: @HoVanDi - Why this is "TodoList" function? Do you still need this file?

File: src/app/Todo.css [hunk 3]
```diff
@@ -0,0 +1,32 @@
+* {
+  margin: 0;
```
Comment (hoihmyit) [inline src/app/Todo.css:2]: @HoVanDi - Do you need this file?

File: src/app/blogs/[id]/page.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
+"use client";
+
+import Card from "react-bootstrap/Card";
+import Button from "react-bootstrap/Button";
```
Comment (hoihmyit) [inline src/app/blogs/[id]/page.tsx]: @HoVanDi - pls remove this line due to it's redundant (it's declared but its value is never used)

File: src/app/blogs/[id]/page.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import Link from "next/link";
+// import useSWR from "swr";
```
Comment (hoihmyit) [inline src/app/blogs/[id]/page.tsx]: Remove this line

File: src/app/blogs/[id]/page.tsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import useSWR, { Fetcher } from "swr";
+const ViewDetailBlog = ({ params }: { params: { id: string } }) => {
+  const fetcher: Fetcher<IBlog, string> = (url: string) =>
+    fetch(url).then((res) => res.json()); //gợi ý cái blog cho mình
+
+  const { data, error, isLoading } = useSWR(
```
Comment (hoihmyit) [inline src/app/blogs/[id]/page.tsx]: `error` variable is not use anymore, so pls remove it

File: src/app/blogs/[id]/page.tsx [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    `http://localhost:8000/blogs/${params.id}`,
```
Comment (hoihmyit) [inline src/app/blogs/[id]/page.tsx]: Try to use the mockAPI as a backend server

File: src/app/blogs/page.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
+"use client";
```
Comment (hoihmyit) [inline src/app/blogs/page.tsx:1]: what does this mean?

File: src/component/app.header.tsx [hunk 9]
```diff
@@ -0,0 +1,33 @@
+"use client";
+import Container from "react-bootstrap/Container";
+import Nav from "react-bootstrap/Nav";
+import Navbar from "react-bootstrap/Navbar";
+import NavDropdown from "react-bootstrap/NavDropdown";
```
Comment (hoihmyit) [inline src/component/app.header.tsx]: Pls remove this line due to it's redundant

File: src/component/app.table.tsx [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+
+const BasicExample = (props: IProps) => {
+  const { blogs } = props; // lấy data
+  console.log(">>check props", blogs);
+
+  const [blog, setBlog] = useState<IBlog | null>(null); // biết được edit tới blog nào
+  const [showModalCreate, setShowModalCreate] = useState<boolean>(false);
+  const [showModalUpdate, setShowModalUpdate] = useState<boolean>(false);
+
+  const handlDeleteBlog = (id: number) => {
```
Comment (hoihmyit) [inline src/component/app.table.tsx]: typo: `handlDeleteBlog` => deleteBlogById

File: package-lock.json [hunk 11]
```diff
@@ -0,0 +1,1848 @@
+{
+  "name": "todolist",
```
Comment (hoihmyit) [inline package-lock.json]: pls rename this variable

File: package-lock.json [hunk 12]
```diff
@@ -0,0 +1,1848 @@
...
+  "version": "0.1.0",
+  "lockfileVersion": 3,
+  "requires": true,
+  "packages": {
+    "": {
+      "name": "todolist",
```
Comment (hoihmyit) [inline package-lock.json]: pls rename this variable

File: src/app/Todo.css (hunk 3 above)
Comment (HoVanDi) [reply src/app/Todo.css:2]: yes

File: src/app/blogs/[id]/page.tsx (hunk 4 above)
Comment (HoVanDi) [reply src/app/blogs/[id]/page.tsx]: Thanks for commenting and I've removed that line

File: src/app/Admin/page.tsx (hunk 2 above)
Comment (HoVanDi) [reply src/app/Admin/page.tsx]: Thanks for commenting and I have changed it

File: src/app/blogs/[id]/page.tsx (hunk 5 above)
Comment (HoVanDi) [reply src/app/blogs/[id]/page.tsx]: Thank you for your feedback and I have removed that line

File: src/app/blogs/[id]/page.tsx (hunk 7 above)
Comment (HoVanDi) [reply src/app/blogs/[id]/page.tsx]: Thank you for your comments and I have used MockAPI as a backend server

File: package.json (hunk 1 above)
Comment (HoVanDi) [reply package.json]: Thank you for your feedback and I have finished making changes

File: src/app/blogs/[id]/page.tsx (hunk 6 above)
Comment (HoVanDi) [reply src/app/blogs/[id]/page.tsx]: Thank you for your response and I have removed the error variable

File: src/component/app.header.tsx (hunk 9 above)
Comment (HoVanDi) [reply src/component/app.header.tsx]: Thank you for your feedback and I have removed that line

File: src/component/app.table.tsx (hunk 10 above)
Comment (HoVanDi) [reply src/component/app.table.tsx]: Thanks for your feedback and I have completed the changes handlDeleteBlog => deleteBlogById

File: package-lock.json (hunk 11 above)
Comment (HoVanDi) [reply package-lock.json]: Thank you for your feedback and I have changed the todolist name

File: package-lock.json (hunk 12 above)
Comment (HoVanDi) [reply package-lock.json]: Thank you for your feedback and I have changed the todolist name

**Comment by hoihmyit** [comment]
image
@HoVanDi - pls update the "Go back" button to more beautiful
And one thing, you should show the modal "View" button as "Edit" button.

File: src/app/blogs/page.tsx (hunk 8 above)
Comment (HoVanDi) [reply src/app/blogs/page.tsx:1]: In Next.js, there is no official hook or syntax for "use client.". So implementing logic on the client-side often involves tasks that cannot be performed on the server-side, such as interacting with the DOM or calling APIs directly from the browser

File: src/app/layout.tsx [hunk 13]
```diff
@@ -0,0 +1,30 @@
+"use client";
+
+import { Inter } from "next/font/google";
+// import './globals.css'
```
Comment (hoihmyit) [inline src/app/layout.tsx]: Pls remove this redundant line

File: src/component/create.modal.tsx [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+  const { showModalCreate, setShowModalCreate } = props;
+  const [title, setTitle] = useState<string>("");
+  const [author, setAuthor] = useState<string>("");
+  const [content, setContent] = useState<string>("");
+
+  const handleSubmit = () => {
+    if (!content && !content && !author) {
+      toast.error("Not empty data !");
+      return;
+    }
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: @HoVanDi - pls move this if the condition to line 38 (behind the `if (!content) {`

File: src/component/create.modal.tsx [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+      .then((res) => res.json())
+      .then((res) => {
+        if (res) {
+          toast.success("Create new blogs succeed");
+          handleCloseModal();
+          mutate("https://652c0736d0d1df5273ef0e4e.mockapi.io/api/v1/blogs"); // gọi lại data để show ra mà không cân reload
+        }
+      });
+    // toast.success("Create succeed!...");
+    // console.log("check data form", title, author, content);
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: remove these line

File: src/component/create.modal.tsx [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+        <Modal.Header closeButton>
+          <Modal.Title>Add new</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form>
+            <Form.Group className="mb-3">
+              <Form.Label>Title</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="....."
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: update to "Enter the title"

File: src/component/create.modal.tsx [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+                value={title}
+                onChange={(e) => setTitle(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group className="mb-3">
+              <Form.Label>Author</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="....."
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: update to "Enter the author"

File: src/component/create.modal.tsx [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+    <>
+      <Modal
+        show={showModalCreate}
+        onHide={() => handleCloseModal()}
+        backdrop="static"
+        keyboard={false}
+        size="lg" // modal dài ra
+      >
+        <Modal.Header closeButton>
+          <Modal.Title>Add new</Modal.Title>
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: Pls update to "Add Blog"

File: src/component/create.modal.tsx [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+                type="text"
+                placeholder="....."
+                value={author}
+                onChange={(e) => setAuthor(e.target.value)}
+              />
+            </Form.Group>
+
+            <Form.Group className="mb-3">
+              <Form.Label>Content</Form.Label>
+              <Form.Control
```
Comment (hoihmyit) [inline src/component/create.modal.tsx:100]: add the placeholder for content. "Enter the content"

File: src/component/update.modal.tsx [hunk 20]
```diff
@@ -0,0 +1,136 @@
...
+      setTitle(blog.title);
+      setAuthor(blog.author);
+      setContent(blog.content);
+    }
+  }, [blog]);
+  const handleSubmit = () => {
+    if (!content && !content && !author) {
+      toast.error("Not empty data !");
+      return;
+    }
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: same as https://github.com/hoihmyit/crud_app/pull/1/files#r1398408327

File: src/component/update.modal.tsx [hunk 21]
```diff
@@ -0,0 +1,136 @@
...
+      .then((res) => res.json())
+      .then((res) => {
+        if (res) {
+          toast.success("Update succeed");
+          handleCloseModal();
+          mutate("https://652c0736d0d1df5273ef0e4e.mockapi.io/api/v1/blogs"); // gọi lại data để show ra mà không cân reload
+        }
+      });
+    // toast.success("Create succeed!...");
+    // console.log("check data form", title, author, content);
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: pls remove these line

File: src/component/update.modal.tsx [hunk 22]
```diff
@@ -0,0 +1,136 @@
...
+        <Modal.Header closeButton>
+          <Modal.Title>Update</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form>
+            <Form.Group className="mb-3">
+              <Form.Label>Title</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="....."
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: with update form, do not keep the placeholder

File: src/component/update.modal.tsx [hunk 23]
```diff
@@ -0,0 +1,136 @@
...
+                value={title}
+                onChange={(e) => setTitle(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group className="mb-3">
+              <Form.Label>Author</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="....."
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: with update form, do not keep the placeholder

File: src/app/layout.tsx (hunk 13 above)
Comment (HoVanDi) [reply src/app/layout.tsx]: Thank you for your feedback and I have removed that line

File: src/component/create.modal.tsx (hunk 14 above)
Comment (HoVanDi) [reply src/component/create.modal.tsx]: Thank you for your feedback. I've moved this if condition to line 38

File: src/component/create.modal.tsx (hunk 15 above)
Comment (HoVanDi) [reply src/component/create.modal.tsx]: Thank you for your feedback and I have removed those line

File: src/component/create.modal.tsx (hunk 16 above)
Comment (HoVanDi) [reply src/component/create.modal.tsx]: Thank you for your feedback and I've updated

File: src/component/create.modal.tsx (hunk 18 above)
Comment (HoVanDi) [reply src/component/create.modal.tsx]: Thank you for your feedback and I've updated

File: src/component/create.modal.tsx (hunk 19 above)
Comment (HoVanDi) [reply src/component/create.modal.tsx:100]: Thank you for your feedback and I have added placeholder

File: src/component/update.modal.tsx (hunk 20 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I have corrected it above

File: src/component/update.modal.tsx (hunk 21 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I have removed those line

File: src/component/update.modal.tsx (hunk 22 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I updated

File: src/component/update.modal.tsx (hunk 23 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I deleted the placeholder

**Review by HoVanDi — COMMENTED** [review]
I finished fixing the error please check. thank you

File: src/component/create.modal.tsx [hunk 24]
```diff
@@ -0,0 +1,123 @@
...
+    if (!title) {
+      toast.error("Not empty title !");
+      return;
+    }
+    if (!author) {
+      toast.error("Not empty author !");
+      return;
+    }
+
+    if (!content && !content && !author) {
```
Comment (hoihmyit) [inline src/component/create.modal.tsx]: @HoVanDi - pls check the if condition here

File: src/component/update.modal.tsx [hunk 25]
```diff
@@ -0,0 +1,133 @@
...
+      return;
+    }
+    if (!author) {
+      toast.error("Not empty author !");
+      return;
+    }
+    if (!content && !content && !author) {
+      toast.error("Not empty data !");
+      return;
+    }
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: @HoVanDi - pls check the if condition here

File: src/component/create.modal.tsx (hunk 16 above)
Comment (hoihmyit) [reply src/component/create.modal.tsx]: @HoVanDi - No, I mean that you should update the "placeholder" value to "Enter the title", not "......."
And keep the Label value as before: "Title"

File: src/component/update.modal.tsx [hunk 26]
```diff
@@ -0,0 +1,133 @@
...
+              <Form.Label>Enter the title</Form.Label>
+              <Form.Control
+                type="text"
+                placeholder="Enter the title"
+                value={title}
+                onChange={(e) => setTitle(e.target.value)}
+              />
+            </Form.Group>
+            <Form.Group className="mb-3">
+              <Form.Label>Enter the Author</Form.Label>
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: @HoVanDi - No, I mean that you should update the "placeholder" value to "Enter the author"
And keep the Label value as before: "Author"

File: src/component/update.modal.tsx [hunk 27]
```diff
@@ -0,0 +1,133 @@
...
+              <Form.Control
+                type="text"
+                value={author}
+                onChange={(e) => setAuthor(e.target.value)}
+              />
+            </Form.Group>
+
+            <Form.Group className="mb-3">
+              <Form.Label>Enter the content</Form.Label>
```
Comment (hoihmyit) [inline src/component/update.modal.tsx]: @HoVanDi - pls update to:
1, <Form.Label>Content</Form.Label>
2, <Form.Control
                as="textarea"
                rows={3}
                placeholder="Enter the content"
                value={content}
                onChange={(e) => setContent(e.target.value)}
              />

File: src/app/blogs/[id]/page.tsx [hunk 28]
```diff
@@ -0,0 +1,54 @@
...
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <div>
+      <div className="my-3">
+        <Link
+          className="btn-back"
+          href={"/blogs"}
```
Comment (hoihmyit) [inline src/app/blogs/[id]/page.tsx:29]: @HoVanDi - pls update this ref due to a bug happen when clicking on the "Blogs" button.
image

File: src/component/create.modal.tsx (hunk 24 above)
Comment (hoihmyit) [reply src/component/create.modal.tsx]: pls remove this if condition

Comment (HoVanDi) [reply src/component/create.modal.tsx]: Thank you for your feedback. But if I remove that condition the program runs incorrectly

File: src/component/update.modal.tsx (hunk 25 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and that condition is correct

File: src/component/update.modal.tsx (hunk 26 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I have updated as you commented

File: src/component/update.modal.tsx (hunk 27 above)
Comment (HoVanDi) [reply src/component/update.modal.tsx]: Thank you for your feedback and I have updated as you commented

File: src/component/create.modal.tsx (hunk 24 above)
Comment (hoihmyit) [reply src/component/create.modal.tsx]: @HoVanDi - so you should update the condition correctly

File: src/component/update.modal.tsx (hunk 25 above)
Comment (hoihmyit) [reply src/component/update.modal.tsx]: @HoVanDi - why the if condition here is correct?
while `!content && !content` is duplicate check for content?

**Review by hoihmyit — APPROVED** [review]

**Pull request merged**